Core/src/org/sleuthkit/autopsy/coreutils/AppSQLiteDB.java

```diff
@@ -0,0 +1,249 @@
+    
+    private final AbstractFile dbAbstractFile;  // AbstractFile for the DB file
+    
+    private Connection connection = null;
+    private Statement statement = null;
+    
+   private AppSQLiteDB(AbstractFile dbAbstractFile, File dbFileCopy) {
+        this.dbAbstractFile = dbAbstractFile;
+        
+        try {
+            Class.forName("org.sqlite.JDBC"); //NON-NLS //load JDBC driver
+            connection = DriverManager.getConnection("jdbc:sqlite:" + dbFileCopy.getPath()); //NON-NLS
+            statement = connection.createStatement();
+        } catch (ClassNotFoundException | SQLException e) {
+            logger.log(Level.SEVERE, "Error opening database " + dbFileCopy.getPath(), e); //NON-NLS
+            connection = null;
+            statement = null;
+        }
+    }
+    
+   
+    /**
+     * Looks for the given SQLIte database filename, with matching path substring. 
+     * It looks for exact name or a pattern match based on 
+     * It makes a copy of each matching file, and creates an instance of 
+     * AppSQLiteDB to help query the DB. 
+     * 
+     * A list of AppSQLiteDB instances is returned, one for each 
+     * match found., 
+     * .
+     * @param dataSource data source to search in 
+     * @param dbName db file name to search
+     * @param matchExactName whether to look for exact file name or a pattern match
+     * @param parentPathSubstr path substring to match
+     * 
+     * @return AbstractFile for the DB if the database file is found.
+     *         Returns NULL if no such database is found.
+     */
+    public static Collection<AppSQLiteDB> findAppDatabases(DataSource dataSource, String dbName, boolean matchExactName, String parentPathSubstr) {
+        
+        List<AppSQLiteDB> appDbs = new ArrayList<> ();
+        Case openCase;
+        
+        try {
+            openCase = Case.getCurrentCaseThrows();
+        } catch (NoCurrentCaseException ex) {
+            Logger.getLogger(AppSQLiteDB.class.getName()).log(Level.SEVERE, "Exception while getting open case.", ex); //NON-NLS
+            return appDbs;
+        }
+        
+        List<AbstractFile> absFiles;
+        long fileId = 0;
+        String localDiskPath = "";
+        try {
+            SleuthkitCase skCase = openCase.getSleuthkitCase();
+            String parentPath = parentPathSubstr.replace("\\", "/");
+            parentPath = SleuthkitCase.escapeSingleQuotes(parentPath);
+            String whereClause;
+            if (matchExactName) {
+                whereClause = String.format("LOWER(name) LIKE LOWER(\'%1$s\') AND LOWER(parent_path) LIKE LOWER(\'%%%2$s%%\') AND data_source_obj_id = " + dataSource.getId(), dbName, parentPath);
+            } else {
+                whereClause = String.format("LOWER(name) LIKE LOWER(\'%%%1$s%%\') AND LOWER(name) NOT LIKE LOWER(\'%%journal%%\') AND LOWER(parent_path) LIKE LOWER(\'%%%2$s%%\') AND data_source_obj_id = " + dataSource.getId(), dbName, parentPath);
```

Comment:
What example did you come across where you needed to wild card the database name?

Reply:
Skype database was named live:btforensics.db (i think) where btforensics is the profile name of the user.

Actually as it turns out, Autopsy did not correctly export that file so it isn't in the logical file data source.